src/settings/global_user.rs

```diff
@@ -17,7 +26,90 @@ pub enum GlobalUser {
 
 impl GlobalUser {
     pub fn new() -> Result<Self, failure::Error> {
-        get_global_config()
+        let environment = Environment::with_whitelist(ENV_VAR_WHITELIST.to_vec());
+
+        let config_path = default_config_file().expect("could not find global config directory");
+
+        GlobalUser::build(environment, config_path)
+    }
+
+    fn build<T: 'static + QueryEnvironment>(
+        environment: T,
+        config_path: PathBuf,
+    ) -> Result<Self, failure::Error>
+    where
+        T: config::Source + Send + Sync,
+    {
+        if let Some(user) = Self::from_env(environment) {
+            user
+        } else {
+            Self::from_file(config_path)
+        }
+    }
+
+    // TODO: This fn should return None if the environment has none of the
+    // relevant variables
+    fn from_env<T: 'static + QueryEnvironment>(
+        environment: T,
+    ) -> Option<Result<Self, failure::Error>>
+    where
+        T: config::Source + Send + Sync,
+    {
+        if environment.empty().unwrap() {
+            None
+        } else {
+            let mut s = config::Config::new();
+            s.merge(environment).ok();
+
+            Some(GlobalUser::from_config(s))
+        }
+    }
+
+    fn from_file(config_path: PathBuf) -> Result<Self, failure::Error> {
+        let mut s = config::Config::new();
+
+        let config_str = config_path
+            .to_str()
+            .expect("global config path should be a string");
+
+        // Skip reading global config if non existent
+        // because envs might be provided
+        if config_path.exists() {
+            log::info!(
+                "Config path exists. Reading from config file, {}",
+                config_str
+            );
+            s.merge(config::File::with_name(config_str))?;
+        } else {
+            failure::bail!("config path does not exist {}", config_str);
```

Comment:
how can the user recover from this? can we give them an actionable?

Reply:
idk, copy pasta.